python/python/lance/util.py

```diff
@@ -108,19 +106,18 @@ def __repr__(self) -> str:
         return f"lance.KMeans(k={self.k}, metric_type={self._metric_type})"
 
     @property
-    def centroids(self) -> Optional[CENTROIDS_TYPE]:
+    def centroids(self) -> Optional[pa.FixedShapeTensorArray]:
         """Returns the centroids of the model,
 
         Returns None if the model is not trained.
         """
         ret = self._kmeans.centroids()
         if ret is None:
             return None
-        if has_fixed_shape_tensor:
-            # Pyarrow compatibility
-            shape = (ret.type.list_size,)
-            tensor_type = pa.fixed_shape_tensor(ret.type.value_type, shape)
-            ret = pa.FixedShapeTensorArray.from_storage(tensor_type, ret)
+
+        shape = (ret.type.list_size,)
+        tensor_type = pa.fixed_shape_tensor(ret.type.value_type, shape)
+        ret = pa.FixedShapeTensorArray.from_storage(tensor_type, ret)
```

Comment:
PyArrow introducted FixedShapeTensor in 13.0.0, and our minimum pyarrow is now 14.0.0. So don't need to handle older versions now.